multi_problem_xblock/multi_problem_xblock.py

```diff
@@ -368,6 +368,13 @@ def student_view_context(self, context=None):
         if overall_progress == 100 and self.current_slide == -1 and self.display_feedback == DISPLAYFEEDBACK.NEVER:
             self.current_slide = 0
 
+        # Clamp current_slide to a valid index when the number of items changes
+        # Keep -1 as a special value to show the test results slide
+        num_items = len(items)
+        if self.current_slide != -1:
+            if self.current_slide < -1 or self.current_slide >= num_items:
+                self.current_slide = 0
```

Comment:
huh I just noticed that this can be (optionally) simplified further:


```suggestion
        if self.current_slide < -1 or self.current_slide >= num_items:
            self.current_slide = 0
```